packages/e2e-tests/test/test-shell.ts

```diff
@@ -222,39 +222,19 @@ export class TestShell {
     return this._process;
   }
 
-  async waitForPrompt(start = 0): Promise<void> {
-    await eventually(() => {
-      const output = this._output.slice(start);
+  /**
+   * Wait for the last line of the output to become a prompt and resolve with it
+   */
+  async waitForPrompt(start = 0): Promise<string> {
+    return this.eventually(() => {
+      const output = this._output.slice(start).trim();
       const lines = output.split('\n');
-      const found = !!lines
-        .filter((l) => PROMPT_PATTERN.exec(l)) // a line that is the prompt must at least match the pattern
-        .find((l) => {
-          // in some situations the prompt occurs multiple times in the line (but only in tests!)
-          const prompts = l
-            .trim()
-            .replace(/>$/g, '')
-            .split('>')
-            .map((m) => m.trim());
-          // if there are multiple prompt parts they must all equal
-          if (prompts.length > 1) {
-            for (const p of prompts) {
-              if (p !== prompts[0]) {
-                return false;
-              }
-            }
-          }
-          return true;
-        });
-      if (!found) {
-        throw new assert.AssertionError({
-          message: 'expected prompt',
-          expected: PROMPT_PATTERN.toString(),
-          actual:
-            this._output.slice(0, start) +
-            '[prompt search starts here]' +
-            output,
-        });
-      }
+      const lastLine = lines[lines.length - 1];
+      assert(
+        PROMPT_PATTERN.test(lastLine),
+        `Expected a prompt (last line was "${lastLine}")`
+      );
+      return lastLine;
```

Comment:
Now returning the prompt since we need this in `executeLine` to omit it from the output.